sdk/monitor/monitor-opentelemetry-exporter/src/export/base.ts

```diff
@@ -0,0 +1,202 @@
+    this._numConsecutiveRedirects = 0;
+    const connectionString = options.connectionString || process.env[ENV_CONNECTION_STRING];
+    this._options = {
+      ...DEFAULT_EXPORTER_CONFIG,
+    };
+    this._options.apiVersion = options.apiVersion ?? this._options.apiVersion;
+    this._options.aadTokenCredential = options.aadTokenCredential;
+
+    if (connectionString) {
+      const parsedConnectionString = ConnectionStringParser.parse(connectionString);
+      this._options.instrumentationKey =
+        parsedConnectionString.instrumentationkey ?? this._options.instrumentationKey;
+      this._options.endpointUrl =
+        parsedConnectionString.ingestionendpoint?.trim() ?? this._options.endpointUrl;
+    }
+    // Instrumentation key is required
+    if (!this._options.instrumentationKey) {
+      const message =
+        "No instrumentation key or connection string was provided to the Azure Monitor Exporter";
+      diag.error(message);
+      throw new Error(message);
+    }
+
+    this._instrumentationKey = this._options.instrumentationKey;
+    this._sender = new HttpSender(this._options);
+    this._persister = new FileSystemPersist(this._options);
+    this._retryTimer = null;
+    diag.debug("AzureMonitorTraceExporter was successfully setup");
+  }
+
+  /**
+   * Persist envelopes to disk
+   */
+  private async _persist(envelopes: unknown[]): Promise<ExportResult> {
+    try {
+      const success = await this._persister.push(envelopes);
+      return success
+        ? { code: ExportResultCode.SUCCESS }
+        : {
+            code: ExportResultCode.FAILED,
+            error: new Error("Failed to persist envelope in disk."),
+          };
+    } catch (ex: any) {
+      return { code: ExportResultCode.FAILED, error: ex };
+    }
+  }
+
+  /**
+   * Shutdown exporter
+   */
+  protected async _shutdown(): Promise<void> {
+    return this._sender.shutdown();
+  }
+
+  /**
+   * Export envelopes
+   */
+  protected async _exportEnvelopes(envelopes: Envelope[]): Promise<ExportResult> {
+    diag.info(`Exporting ${envelopes.length} envelope(s)`);
+
+    try {
+      const { result, statusCode } = await this._sender.send(envelopes);
+      this._numConsecutiveRedirects = 0;
+      if (statusCode === 200) {
+        // Success -- @todo: start retry timer
+        if (!this._retryTimer) {
+          this._retryTimer = setTimeout(() => {
+            this._retryTimer = null;
+            this._sendFirstPersistedFile();
+          }, this._options.batchSendRetryIntervalMs);
+          this._retryTimer.unref();
+        }
+        return { code: ExportResultCode.SUCCESS };
+      } else if (statusCode && isRetriable(statusCode)) {
+        // Failed -- persist failed data
+        if (result) {
+          diag.info(result);
+          const breezeResponse = JSON.parse(result) as BreezeResponse;
+          const filteredEnvelopes: Envelope[] = [];
+          breezeResponse.errors.forEach((error) => {
```

Comment:
Would breezeResponse.errors be populated if response is not 206? Would this error out if `errors` is `None`?